src/Chainweb/Mempool/InMem.hs

```diff
@@ -721,13 +721,14 @@ pruneInternal
 pruneInternal mdata now = do
     let pref = _inmemPending mdata
     !pending <- readIORef pref
-    let !pending' = HashMap.filter flt pending
+    !pending' <- evaluate $ force $ HashMap.filter flt pending
```

Comment:
Again, this feels like hitting a cockroach on the head three times. `force` does all the work that it's possible to do here.